File: languages/python/bloock/entity/availability/ipfs_loader.py
import bloock._bridge.proto.availability_entities_pb2 as proto
from bloock.entity.availability.loader import Loader
from bloock.entity.availability.loader_args import LoaderArgs


class IpfsLoader(Loader):
    """
    Represents a loader for IPFS data availability.
    """
    def __init__(self, hash: str) -> None:
        """
        Constructs a IpfsLoader object with the specified parameters.
        :type hash: object
        :rtype: object
        """
        super().__init__(type=proto.IPFS, args=LoaderArgs(hash))

    def to_proto(self) -> proto.Loader:
        return proto.Loader(type=self.type, args=self.args.to_proto())

File: languages/php/Bloock/Entity/Availability/IpfsLoader.php
<?php

namespace Bloock\Entity\Availability;

use Bloock\DataAvailabilityType;

/**
 * Represents a loader for IPFS data availability.
 */
class IpfsLoader implements Loader
{
    private int $type;
    private LoaderArgs $args;

    /**
     * Constructs a IpfsLoader object with the specified parameters.
     * @param string $id
     */
    public function __construct(string $id)
    {
        $this->type = DataAvailabilityType::IPFS;
        $this->args = new LoaderArgs($id);
    }

    public function toProto(): \Bloock\Loader
    {
        $p = new \Bloock\Loader();
        $p->setType($this->type);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}